### PR TITLE
Change KerasClassifier to previous behavior

### PR DESCRIPTION
**Pull request by beat-buesser**
Signed-off-by: Beat Buesser <beat.buesser@ie.ibm.com>

# Description

The recently merged pull request #119 has unintendedly changed the behaviour of `KerasClassifier` with regards to `use_logits` if the loss function is  `categorical_crossentropy`. The change is visible in `adversarial-training-mnist.ipynb`. We have identified the cause at the different implementations of `categorical_crossentropy` in keras (https://github.com/keras-team/keras/blob/c10d24959b0ad615a21e671b180a1b2466d77a2b/keras/backend/tensorflow_backend.py#L3527) and tensorflow.keras (https://github.com/tensorflow/tensorflow/blob/456fbc0e498e3d10604973de9f46ca48d62267cc/tensorflow/python/keras/backend.py#L4120). It seems that the previous version of `KerasClassifier` on `master`has been accounting for this without explicitly mentioning it at https://github.com/IBM/adversarial-robustness-toolbox/blob/f49fd6caf2fcf815c17dcc611ccf11cc134003ac/art/classifiers/keras.py#L135.

This pull request adds back the previous behaviour of `KerasClassifier` to only follow the approach of tensorflow.keras with `categorical_crossentropy`.

## Type of change

Please check all relevant options.

- [x] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by beat-buesser** [comment]
Hi @ririnicolae Thank you very much! The tests are a great suggestion. I'll add them soon to this PR.
